core/templates/pages/contributor-dashboard-page/services/translate-text-backend-api.service.ts

```diff
@@ -39,13 +39,24 @@ export class TranslateTextBackendApiService {
       return TranslatableTexts.createFromBackendDict(backendDict);
     });
   }
-
+  async blobtoBase64(blob: Blob): Promise<string> {
+    return new Promise<string> ((resolve, reject)=> {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        const dataurl = reader.result as string;
+        const base64 = dataurl.substring(dataurl.indexOf(',') + 1);
```

Comment:
Could you please add a comment explaining what is going on here? It seems a bit weird to have logic like this inside a standard conversion function.

Reply:
Done.

Reply:
Please describe a bit more what you mean by "prefix". Is this always a fixed string? If so, maybe an exception should be thrown if the starting characters are not what is expected?

Reply:
Can you do this along the lines of: "if dataurl startswith 'data:mime/type;base64,', remove the prefix, otherwise throw an error"?

(Is this prefix the only possible prefix here? If not, should account for all possibilities in the above-suggested approach.)

Reply:
> (Is this prefix the only possible prefix here? If not, should account for all possibilities in the above-suggested approach.)

Yes, this is the only possible prefix where type varies on image data for which I have covered all cases and have written a test case to prove corresponding changes.

Reply:
Thanks -- new approach looks good. It would be a little better if you can extract the exact matching string (when doing the regex comparison), find its length, and remove that substring instead, so that you don't have a "magic constant" here (in the form of the ','). Otherwise the lines of code don't connect to each other so well, because it's not clear where the ',' comes from.

Reply:
Thanks. I think instead of proposed approach we can use `replace()`
```suggestion
        const base64 = dataurl.replace(prefixRegex, '');
```
to extract base64 encoded string.

Reply:
Thanks. I think instead of proposed approach we can use `replace()`
```suggestion
        const base64 = dataurl.replace(prefixRegex, '');
```
to extract base64 encoded string.